### ISSUE TITLE
[Bug]: NextLink not showing redirect links when using keyboard controls

### ISSUE THREAD

**Issue by maximegheraille** [tailwindlabs/headlessui#653]
### What package within Headless UI are you using?

@headlessui/react

### What version of that package are you using?

1.3.0

### What browser are you using?

chrome

### Reproduction repository

https://owvp3.sse.codesandbox.io

### Describe your issue

Hi,

I got the same problem as this bug report #20 and made it work wit the [solutions](https://github.com/tailwindlabs/headlessui/issues/20#issuecomment-701482746) in the comments but it creates a new issue.

When using the keyboard controls to navigatie a `<Menu.Item>` that contains a `<Link>` the navigator does not show that it is redirects to another page. 
We should have this for example this: <img width="88" alt="Screen Shot 2021-06-30 at 20 52 33" src="https://user-images.githubusercontent.com/32335849/124015875-59aa4000-d9e5-11eb-93cc-bb3bec85fc3b.png"> at the bottom left of the screen.

It does work when hovering on it but not when using the keyboard controls.

any idea on how to fix it?

have a nice day

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

The reason you don't see it is because the actual `a` link is not focused when using the keyboard. The wrapper maintains the focus so that it is focus trapped and you can't escape it.
The reason it works for hover is because that's the default browser behaviour.

We could implement the `roving tab index` style but that's a big change in all our components.

Long story short, currently this is not possible.
